mypy/main.py

```diff
@@ -381,6 +382,15 @@ def add_invertible_flag(flag: str,
     # --local-partial-types disallows partial types spanning module top level and a function
     # (implicitly defined in fine-grained incremental mode)
     parser.add_argument('--local-partial-types', action='store_true', help=argparse.SUPPRESS)
+    # --bazel changes some behaviors for use with Bazel (https://bazel.build).
+    parser.add_argument('--bazel', action='store_true', help=argparse.SUPPRESS)
+    # --cache-map FILE ... gives a mapping from source files to cache files.
```

Comment:
I am interested in what happened to the idea of dropping cache files next to source files. I remember choosing the same option you did here, but I forget why, because now-me had the idea that cache files next to source would be cleaner for some reason.

Reply:
I tried to write this up in my overly long status report in #3912 -- Dropbox's build environment puts the output files in a different directory than the source files. It seems outputs (and inputs) must be `File` objects, the only way to create those is `ctx.actions.declare_file()`, which always puts it it in the output directory. (The `File` objects for the sources come from `ctx.files.srcs`, but I don't think I can add new ones via that approach.)

Reply:
That is correct -- you cannot add anything via ctx.files.srcs. You should assume that the source tree is read-only.

If you declare a file, you must create it within the action (Bazel checks this).